### PR TITLE
Remove OnRuntimeUpgrade hooks

### PR DESCRIPTION
**Pull request by Dengjianping**
Signed-off-by: Dengjianping <djptux@gmail.com>

## Description

Remove `OnRuntimeUpgrade` hooks after v3.2.0 has been released.

closes: paritytech/substrate#613 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests.
- [ ] Updated relevant documentation in the code.
- [x] Added **one** line describing your change in `<branch>/CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] If runtime changes, need to update the version numbers properly:
   * `authoring_version`: The version of the authorship interface. An authoring node will not attempt to author blocks unless this is equal to its native runtime.
   * `spec_version`: The version of the runtime specification. A full node will not attempt to use its native runtime in substitute for the on-chain Wasm runtime unless all of spec_name, spec_version, and authoring_version are the same between Wasm and native.
   * `impl_version`: The version of the implementation of the specification. Nodes are free to ignore this; it serves only as an indication that the code is different; as long as the other two versions are the same then while the actual code may be different, it is nonetheless required to do the same thing. Non-consensus-breaking optimizations are about the only changes that could be made which would result in only the impl_version changing.
   * `transaction_version`: The version of the extrinsics interface. This number must be updated in the following circumstances: extrinsic parameters (number, order, or types) have been changed; extrinsics or pallets have been removed; or the pallet order in the construct_runtime! macro or extrinsic order in a pallet has been changed. You can run the `metadata_diff.yml` workflow for help. If this number is updated, then the `spec_version` must also be updated 
- [ ] Verify benchmarks & weights have been updated for any modified runtime logics
- [ ] If importing a new pallet, choose a proper module index for it, and allow it in `BaseFilter`. Ensure **every** extrinsic works from front-end. If there's corresponding tool, ensure both work for each other.
- [ ] If needed, update our Javascript/Typescript APIs. These APIs are officially used by exchanges or community developers.
- [ ] If modifying existing runtime storage items, make sure to implement storage migrations for the runtime and test them with `try-runtime`. This includes migrations inherited from upstream changes, and you can search the diffs for modifications of `#[pallet::storage]` items to check for any.

### CODE REVIEW

**Comment by Garandor** [comment]
is this a draft or ready for review?

**Comment by Garandor** [comment]
The idea of using the StorageVersion guards is supposed to enable keeping the migration code in the pallet after migration. 
See https://github.com/paritytech/polkadot-sdk/issues/343

I think we should adopt this best practice and keep executed migrations in the pallet as a sort of "backlog"

**Comment by ghzlatarev** [comment]
> The idea of using the StorageVersion guards is supposed to enable keeping the migration code in the pallet after migration. See [paritytech/polkadot-sdk#343 (comment)](https://github.com/paritytech/polkadot-sdk/issues/343)
> 
> I think we should adopt this best practice and keep executed migrations in the pallet as a sort of "backlog"

Yeah. When we need to use the pallet hook again we can just copy paste the old migration to some separate file, or we can do it now.

**Review by Garandor — CHANGES_REQUESTED** [review]
pls re-request when ready

**Comment by Dengjianping** [comment]
> 

Now it's ready.

**Review by Garandor — CHANGES_REQUESTED** [review]
Could you pls move the migration from the `asset-manager` src to a `migrations` submodule?
We should keep those separate from the active code.
Check `manta-collator-selection` for reference

**Review by ghzlatarev — DISMISSED** [review]

**Review by ghzlatarev — DISMISSED** [review]

**Comment by Dengjianping** [comment]
> Could you pls move the migration from the `asset-manager` src to a `migrations` submodule? We should keep those separate from the active code. Check `manta-collator-selection` for reference

Done.

**Review by Garandor — DISMISSED** [review]
LGTM!

**Comment by ghzlatarev** [comment]
@Dengjianping Merge conflicts again :(
@Garandor we'll need to re-review this when resolved...

**Review by ghzlatarev — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,10 +8,11 @@
 ### Features
 
 ### Improvements
-- [\#607](https://github.com/Manta-Network/Manta/pull/607) Turn node client code into library for CLI project
 - [\#449](https://github.com/Manta-Network/Manta/pull/449) Remove strip from CI, and add strip profile to production.
 - [\#576](https://github.com/Manta-Network/Manta/pull/576) Unfilter xtokens.transfer_multicurrencies and bump MaxAssetsForTransfer to 2.
-- [\#609](https://github.com/Manta-Network/Manta/pull/609) Update parameter path from `sdk` to `manta-parameters`
```
Comment (Garandor) [inline CHANGELOG.md]: Found a mistake with the log. Seems i cant add commits to this branch for some reason so i have to leave it here like this. The 609 entry should be 

- [\#622](https://github.com/Manta-Network/Manta/pull/622) Update parameter path from `sdk` to `manta-parameters`.

**Review by Garandor — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (Dengjianping) [reply CHANGELOG.md]: Updated

**Review by ghzlatarev — APPROVED** [review]

**Review by Garandor — APPROVED** [review]

**Pull request merged**